tests/test_coverage.py

```diff
@@ -209,10 +221,15 @@ def test_process_item_succeeds_when_optional_provider_fails(self):
 
         result = self.coverage_provider.process_item(self.identifier)
 
-        # A successful result is achieved.
+        # A successful result is achieved, even though the optional
+        # coverage provider failed.
         eq_(result, self.identifier)
-        # Even though the coverage provider failed and an appropriate
-        # coverage record was created to mark the failure.
-        r = get_one(self._db, CoverageRecord, identifier=self.identifier)
-        eq_("What did you expect?", r.exception)
 
+        # An appropriate coverage record was created to mark the failure.
+        presentation_edition = DataSource.lookup(
+            self._db, DataSource.PRESENTATION_EDITION
+        )
+        r = self._db.query(CoverageRecord).filter(
```

Comment:
Because we're attempting to create a work in process_item now, some additional DataSource.PRESENTATION_EDITION coverage records are created.